model/user/user.go

```diff
@@ -184,6 +184,37 @@ func (u *DBUser) IncPatchNumber() (int, error) {
 	return dbUser.PatchNumber, nil
 }
 
+func (u *DBUser) AddRole(role string) error {
+	if util.StringSliceContains(u.SystemRoles, role) {
+		return errors.Errorf("cannot add duplicate role '%s'", role)
+	}
+	update := bson.M{
+		"$push": bson.M{RolesKey: role},
+	}
+	if err := UpdateOne(bson.M{IdKey: u.Id}, update); err != nil {
```

Comment:
Is it conceivable `u.Id` might not exist in the db?

Reply:
I don't think it's a scenario that needs to be handled. Our user model creates a user when attempting to retrieve a user that is not in the db